tests/x11/wireshark.pm

```diff
@@ -45,28 +46,68 @@ sub run {
     if (check_var("VERSION", "Tumbleweed")) {
         type_string "wireshark-gtk\n";
     }
-    else {    # works for SLE12SP1 & SLE12SP2
+    else {    # works for SLE12-SP1+
         type_string "wireshark\n";
     }
     assert_screen "wireshark-welcome", 30;
     send_key "super-up";
     assert_screen "wireshark-fullscreen";
 
+    # check GUI version
+    assert_and_click "wireshark-help";
+    assert_and_click "about-wireshark";
+    assert_screen [qw(wireshark-gui-qt wireshark-gui-gtk)];
+    if (match_has_tag "wireshark-gui-gtk") {
+        $wireshark_gui_version = "gtk";
+    }
+    elsif (match_has_tag "wireshark-gui-qt") {
+        $wireshark_gui_version = "qt";
+    }
+    else {
+        record_info('fail', "Could not determine GUI version");
```

Comment:
in case you fail to detect version . Does it really make sense to continue ? Most of check would be muted , right ?

Reply:
could you please in your local copy change needle tag name to wrong one and make test run to see what test is actually doing when version is not detected ?

Reply:
This can not even be happening because the above assert_screen ensures that one of the two tags has to match. I recommend to just use the ternary operator:

```perl
my $wireshark_gui_version = match_has_tag 'wireshark-gui-qt' ? 'qt' : 'gtk';
```

replacing the whole if/elsif/else

Reply:
This is actually an oversight; I originally used a `check_screen` on line 58 instead of the `assert_screen` because I thought that the `record_info` message on line 67 would have been a useful hint for triaging potential failures. Then I changed it to `assert_screen` and did not remove it. I've amended my commit now.